### DISCUSSION TITLE
Is there a way to check if `prefer_gpu` has been called?

### DISCUSSION THREAD

**Discussion by nikita-galileo** [explosion/spaCy#11648] [Help: Coding & Implementations] [answered]
Trying to understand how we might know if a user using spacy has called `spacy.prefer_gpu` yet. Looking through the thinc code it seems like I could almost check the value of some variable, but I'm not sure. Would love to do a read-only check of if spacy will use the GPU before starting training.

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @nikita-galileo , there's no variable to check. Instead, you can try the [`thinc.api.get_current_ops` function](https://thinc.ai/docs/api-backends#get_current_ops) and check if the class is not an instance of `NumpyOps`. The best practice though is to just call `prefer_gpu` whenever necessary (or if you're uncertain).

**Reply by nikita-galileo** [reply]
Thank you!
